### PR TITLE
Improve my account view for readonly users

### PR DESCRIPTION
**Pull request by dasuni-30**
### Related Issues
- None

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx [hunk 1]
```diff
@@ -62,7 +63,7 @@ export const ProfileWidget: FunctionComponent<ProfileWidgetPropsInterface> = (
     };
 
     const profileCompletion: ProfileCompletion = useSelector((state: AppState) => state.profile.completion);
-
+    const isReadOnlyUser = useSelector((state: AppState) => state.authenticationInformation.profileInfo.isReadOnly);
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/widgets/profile-widget.tsx:66]:
```suggestion
    const isReadOnlyUser: boolean = useSelector((state: AppState) => state.authenticationInformation.profileInfo.isReadOnly);
```

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx [hunk 2]
```diff
@@ -118,27 +119,35 @@ export const ProfileWidget: FunctionComponent<ProfileWidgetPropsInterface> = (
...
                 data-testid={ `${testId}-settings-section` }
                 header={ t("myAccount:components.overview.widgets.profileStatus.header" ,
                     { productName: config.ui.productName }) }
                 description={
-                    <>
+                    (<>
                         {
                             userSource &&
-                            <div
+                            (<div
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/widgets/profile-widget.tsx:134]:
```suggestion
                            userSource && (
                                <div
```

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx [hunk 3]
```diff
@@ -118,27 +119,35 @@ export const ProfileWidget: FunctionComponent<ProfileWidgetPropsInterface> = (
...
                                 className="overview-page-header"
                             >
                                 {  t("myAccount:components.overview.widgets.profileStatus.userSourceText" ,
                                     { source: userSource }) }
-                            </div>
+                            </div>)
+                        }
+                        { !CommonUtils.isProfileReadOnly(isReadOnlyUser)
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/widgets/profile-widget.tsx:141]:
```suggestion
                        {
                            !CommonUtils.isProfileReadOnly(isReadOnlyUser) 
```

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx [hunk 4]
```diff
@@ -118,27 +119,35 @@ export const ProfileWidget: FunctionComponent<ProfileWidgetPropsInterface> = (
...
                         }
-                        { generateCompletionProgress() }
-                    </>
+                    </>)
                 }
-                primaryAction={ "Manage your profile" }
+                primaryAction={ 
+                    !CommonUtils.isProfileReadOnly(isReadOnlyUser) 
+                        ? t("myAccount:components.overview.widgets.profileStatus.description")
+                        : t("myAccount:components.overview.widgets.profileStatus.readOnlyDescription") }
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/widgets/profile-widget.tsx:150]:
```suggestion
                        : t("myAccount:components.overview.widgets.profileStatus.readOnlyDescription")
                    }
```

**Review by brionmario — APPROVED** [review]

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx (hunk 2 above)
Comment (dasuni-30) [reply apps/myaccount/src/components/overview/widgets/profile-widget.tsx]: Fixed by cd03835fb9a6fbe55320390334bb76f31d83f18d.

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx (hunk 4 above)
Comment (dasuni-30) [reply apps/myaccount/src/components/overview/widgets/profile-widget.tsx]: Fixed by cd03835fb9a6fbe55320390334bb76f31d83f18d.

File: apps/myaccount/src/pages/personal-info.tsx [hunk 5]
```diff
@@ -97,6 +100,13 @@ const PersonalInfoPage:  FunctionComponent<PersonalInfoPagePropsInterface> = (
                         : t("myAccount:pages.personalInfoWithoutExportProfile.subTitle") 
             }
         >
+            {
+                CommonUtils.isProfileReadOnly(isReadOnlyUser) 
+                && (<Message
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply apps/myaccount/src/pages/personal-info.tsx]: Fixed by cd03835fb9a6fbe55320390334bb76f31d83f18d.

**Pull request merged**